compiler/sigmatch.nim

```diff
@@ -2506,9 +2451,6 @@ proc matches*(c: PContext, n, nOrig: PNode, m: var TCandidate) =
     m.call = n
     # Note the following doesn't work as it would produce ambiguities.
```

Comment:
this comment now makes no sense after the code is deleted; maybe enhance the comment instead